File: TapTrustSwift/Tests/TapTrustSwiftTests/test_cbor.swift
import XCTest
import SwiftCBOR
@testable import TapTrustSwift

final class CBORTests: XCTestCase {
    func testDecodePayloadReturnsExpectedKeys() throws {
        let payload: [String: Any] = ["foo": "bar", "num": 123]
        let bytes = try CBOR.encodeAny(payload)
        let data = Data(bytes)
        let result = try CBORCOSE.decodePayload(raw: data)
        XCTAssertEqual(result["foo"] as? String, "bar")
        XCTAssertEqual(result["num"] as? Int, 123)
    }
}

File: tests/test_cbor.py
import pytest

from verifier.decode.cbor import decode_payload


def test_decode_payload_returns_expected_keys(tmp_path):
    """CBOR payload bytes should decode into a dictionary"""
    payload = {"foo": "bar", "num": 123}
    try:
        import cbor2
    except ImportError:
        pytest.skip("cbor2 not installed")

    raw = cbor2.dumps(payload)

    result = decode_payload(raw)
    assert isinstance(result, dict)
    assert set(result.keys()) == set(payload.keys())
